content/blog/copilot-lessons/index.md

```diff
@@ -0,0 +1,126 @@
+
+![Example PromptFoo Eval](promptfoo.png)
+
+The eval suite keeps getting more robust, which means that when new AI models drop, we can quickly catch any weirdness before it hits production. The generative AI space moves crazy fast and the code changes a lot, but the evals are a safety net - catching hallucinations, maintaining quality, and making sure we don't ship anything that'll annoy our users.
+
+So, while hallucinations are now much rarer, what about that one with the `--force` flag? Yes, it's "just another bug," but it taught us something fascinating about these AI errors.
+
+## LLMs think like humans (sort of)
+
+The `--force` hallucination wasn't totally wrong - it was revealing what users intuitively expect from the CLI, and the LLM accidentally showed us what was missing. Force deletion is a common pattern across developer tools, and the LLM, trained on vast amounts of documentation and code, simply reflects these established conventions.
+
+This has fundamentally changed how we view hallucinations. While the team works constantly to minimize them – and our eval work means they happen way less frequently – some of them are clearly product signals. The LLM, in this light, becomes an unexpected source of user research, drawing on its training across thousands of developer tools and experiences.
+
+This insight is one of the key lessons of building Copilot:
+
+1. **Minimize LLM Usage:** Let traditional code handle deterministic tasks, reserve LLMs for natural language work
+2. **Decompose into Skills:** Break complex tasks into modular units that combine LLM and traditional code appropriately
+3. **Test Rigorously:** Use multiple validation approaches, including LLMs testing LLMs
+4. **Learn from Hallucinations:** Sometimes incorrect outputs reveal user expectations
+5. **Learn from Users Continuously:** User interactions improve our AI systems - from training better skills to catching hallucinations and revealing product opportunities.
+
+These lessons helped shape our latest release: **[the Pulumi Copilot REST API](/blog/pulumi-copilot-rest/)**, now available in preview. You can integrate these same capabilities and skills into your own tools and workflows. Whether you're building CLI extensions, chat integrations, or automated deployment checks, the API provides a contextual understanding of Copilot. **[Try it out](docs.pulumi.com/Copilot-api).**
```

Comment:
"Try it out" is a broken link. Did you mean to link to the docs here: https://www.pulumi.com/docs/pulumi-cloud/copilot/api/